OSpackages.sh

```diff
@@ -0,0 +1,399 @@
+         ;;
+      *)
+         echo "Unknown OS. Aborting" >&2
+         exit 1
+         ;;
+   esac
+
+   ( IFS=',' ; echo "${BASEREPOS[*]}" )
+}
+
+# Install base/setup packages in chroot-dev
+function PrepChroot {
+   local -a BASEPKGS
+
+   # Create an array of packages to install
+   mapfile -t BASEPKGS < <(
+      rpm --qf '%{name}\n' -qf /etc/os-release ; \
+      rpm --qf '%{name}\n' -qf  /etc/yum.repos.d/* 2>&1 | grep -v "not owned" | sort -u ; \
+      echo yum-utils
+   )
+
+   # Ensure DNS lookups work in chroot-dev
+   if [[ ! -e ${CHROOTMNT}/etc/resolv.conf ]]
+   then
+      err_exit "Installing ${CHROOTMNT}/etc/resolv.conf..." NONE
+      install -Dm 000644 /etc/resolv.conf "${CHROOTMNT}/etc/resolv.conf"
+   fi
+
+   # Ensure etc/rc.d/init.d exists in chroot-dev
+   if [[ ! -e ${CHROOTMNT}/etc/rc.d/init.d ]]
+   then
+      install -dDm 000755 "${CHROOTMNT}/etc/rc.d/init.d"
+   fi
+
+   # Ensure etc/init.d exists in chroot-dev
+   if [[ ! -e ${CHROOTMNT}/etc/init.d ]]
+   then
+      ln -t "${CHROOTMNT}/etc" -s ./rc.d/init.d
+   fi
+
+   # Clean out stale RPMs
+   if [[ $( stat /tmp/*.rpm > /dev/null 2>&1 )$? -eq 0 ]]
+   then
+      err_exit "Cleaning out stale RPMs..." NONE
+      rm -f /tmp/*.rpm || \
+        err_exit "Failed cleaning out stale RPMs"
+   fi
+
+   # Stage our base RPMs
+   yumdownloader --destdir=/tmp "${BASEPKGS[@]}"
+   if [[ ${REPORPMS:-} != '' ]]
+   then
+      FetchCustomRepos
+   fi
+
+   # Initialize RPM db in chroot-dev
+   err_exit "Initializing RPM db..." NONE
+   rpm --root "${CHROOTMNT}" --initdb || \
+     err_exit "Failed initializing RPM db"
+
+   # Install staged RPMs
+   err_exit "Installing staged RPMs..." NONE
+   rpm --root "${CHROOTMNT}" -ivh --nodeps /tmp/*.rpm || \
+     err_exit "Failed installing staged RPMs"
+
+   # Install dependences for base RPMs
+   err_exit "Installing base RPM's dependences..." NONE
+   yum --disablerepo="*" --enablerepo="${OSREPOS}" \
+      --installroot="${CHROOTMNT}" -y reinstall "${BASEPKGS[@]}" || \
+     err_exit "Failed installing base RPM's dependences"
+
+   # Ensure yum-utils are installed in chroot-dev
+   err_exit "Ensuring yum-utils are installed..." NONE
+   yum --disablerepo="*" --enablerepo="${OSREPOS}" \
+      --installroot="${CHROOTMNT}" -y install yum-utils || \
+     err_exit "Failed installing yum-utils"
+
+}
+
+# Install selected package-set into chroot-dev
+function MainInstall {
+   local YUMCMD
+
+   YUMCMD="yum --nogpgcheck --installroot=${CHROOTMNT} "
+   YUMCMD+="--disablerepo=* --enablerepo=${OSREPOS} install -y "
+
+   # Stub...
+   echo "${RPMFILE}" > /dev/null 2>&1
+
+   # Expand the "core" RPM group and store as array
+   mapfile -t INLCLUDEPKGS < <(
+      yum groupinfo "${RPMGRP}" 2>&1 | \
+      sed -n '/Mandatory/,/Optional Packages:/p' | \
+      sed -e '/^ [A-Z]/d' -e 's/^[[:space:]]*[-=+[:space:]]//'
+   )
+
+   # Add extra packages to include-list (array)
+   INLCLUDEPKGS+=( "${INLCLUDEPKGS[@]}" "${MINXTRAPKGS[@]}" )
```

Comment:
these are both arrays right? can't this be simply:
```suggestion
   INLCLUDEPKGS+=MINXTRAPKGS
```

Reply:
Obsoluted by commit